### ISSUE TITLE
Default gRPC max buffer size too small

### ISSUE THREAD

**Issue by altho1** [cloudquery/cloudquery#4834]
### Describe the Bug
Running a sync on a fair sized organisation, I get this error:
"Error: failed to sync source aws: failed to fetch resources from stream: rpc error: code = ResourceExhausted desc = grpc: received message larger than max (9090474 vs. 4194304)" 

Suggest the gRPC default max buffer size is either exposed as a parameter or raised to e.g. 100mb

### Expected Behavior
No resource limit encountered

### CloudQuery Version
cloudquery version 2.0.18

### Debug Output
It is huge - let me know if it is needed
"Error: failed to sync source aws: failed to fetch resources from stream: rpc error: code = ResourceExhausted desc = grpc: received message larger than max (9090474 vs. 4194304)" 

### Steps to Reproduce
```
cloudquery sync .
```

### Additional Context
-

### References
-

**Comment by hermanschaaf** [comment]
Hi @altho1, thanks for reporting this! We will take a look. You may very well be right that we need to increase the default buffer size or expose it as a configuration option.

Out of interest, what source plugin and version are you using? (`aws`, `gcp`, `azure`, etc?) And can you tell from the log output which specific resource was being fetched at the time of the error? I wouldn't generally expect a single resource to exceed 4MB, so we may need to look at the source plugin in question as well.

**Comment by altho1** [comment]
Hi @hermanschaaf,

It is th AWS source plugin (failed to sync source aws), and it is version:
``` yaml
kind: source
spec:
  name: aws
  path: cloudquery/aws
  version: "v6.0.0"
  tables: ["*"]
  destinations: ["postgresql"]
```
The last line before the error states: "- Syncing resources... (456744/-, 1036 resources/s) [11m8s]"
I will add debug and have a look and let you know which resource type it is.

**Comment by hermanschaaf** [comment]
@altho1 Thanks a lot :bow: I'm working on a fix for this from the SDK side, will keep you posted.

**Comment by altho1** [comment]
The debug requested
Here issues start:
```
2022-11-21T13:37:08Z INF fetch table finished client=119311:eu-west-3 errors=0 module=aws-src resources=0 table=aws_iam_user_attached_policies
2022-11-21T13:37:08Z INF table resolver started client=119311:eu-west-3 module=aws-src table=aws_iam_user_policies
2022-11-21T13:37:08Z INF fetch table finished client=182333:ap-south-1 errors=0 module=aws-src resources=1 table=aws_lambda_function_versions
2022-11-21T13:37:08Z INF table resolver started client=182333:ap-south-1 module=aws-src table=aws_lambda_function_concurrency_configs
2022-11-21T13:37:08Z ERR failed to wait error="failed to execute batch: context canceled" module=cli
2022-11-21T13:37:08Z ERR finished call grpc.code=Unknown grpc.component=server grpc.error="context canceled" grpc.method=Write2 grpc.method_type=client_stream grpc.service=proto.Destination grpc.start_time=2022-11-21T13:26:06Z grpc.time_ms=662379.94 module=cli peer.address=@ protocol=grpc
2022-11-21T13:37:08Z ERR column resolver finished with error error="operation error IoT: ListTagsForResource, https response error StatusCode: 0, RequestID: , canceled, context canceled" client=784246:ap-southeast-2 module=aws-src table=aws_iot_jobs
2022-11-21T13:37:08Z ERR column resolver finished with error error="operation error IoT: ListTagsForResource, https response error StatusCode: 0, RequestID: , canceled, context canceled" client=459824:ap-southeast-2 module=aws-src table=aws_iot_thing_groups
2022-11-21T13:37:08Z ERR column resolver finished with error error="operation error IAM: ListPolicyTags, request canceled, context canceled" client=323930:ap-northeast-2 module=aws-src table=aws_iam_policies
2022-11-21T13:37:08Z ERR table resolver finished with error error="operation error Lambda: ListProvisionedConcurrencyConfigs, https response error StatusCode: 0, RequestID: , canceled, context canceled" client=182333:ap-south-1 module=aws-src table=aws_lambda_function_concurrency_configs
2022-11-21T13:37:08Z INF fetch table finished client=182333:ap-south-1 errors=1 module=aws-src resources=0 table=aws_lambda_function_concurrency_configs
2022-11-21T13:37:08Z INF table resolver started client=182333:ap-south-1 module=aws-src table=aws_lambda_function_event_source_mappings
2022-11-21T13:37:08Z ERR table resolver finished with error error="operation error Lambda: ListEventSourceMappings, context canceled" client=182333:ap-south-1 module=aws-src table=aws_lambda_function_event_source_mappings
2022-11-21T13:37:08Z INF fetch table finished client=182333:ap-south-1 errors=1 module=aws-src resources=0 table=aws_lambda_function_event_source_mappings
```
and it continues until it all ends:
```
2022-11-21T13:37:09Z ERR pre resource resolver failed error="operation error IAM: GetRole, request canceled, context canceled" client=521949:ap-northeast-3 module=aws-src table=aws_iam_roles
2022-11-21T13:37:09Z ERR column resolver finished with error error="operation error IoT: ListThingsInThingGroup, https response error StatusCode: 0, RequestID: , canceled, context canceled" client=784246:ap-northeast-2 module=aws-src table=aws_iot_thing_groups
2022-11-21T13:37:09Z ERR column resolver finished with error error="operation error IoT: ListAttachedPolicies, context canceled" client=784246:ap-northeast-2 module=aws-src table=aws_iot_thing_groups
2022-11-21T13:37:09Z ERR column resolver finished with error error="operation error IoT: ListTagsForResource, context canceled" client=784246:ap-northeast-2 module=aws-src table=aws_iot_thing_groups
2022-11-21T13:37:09Z ERR column resolver finished with error error="operation error IoT: ListTagsForResource, https response error StatusCode: 0, RequestID: , canceled, context canceled" client=459824:ap-northeast-2 module=aws-src table=aws_iot_jobs
2022-11-21T13:37:09Z ERR column resolver finished with error error="operation error IoT: ListThingsInThingGroup, https response error StatusCode: 0, RequestID: , canceled, context canceled" client=459824:eu-central-1 module=aws-src table=aws_iot_thing_groups
2022-11-21T13:37:09Z ERR column resolver finished with error error="operation error IoT: ListAttachedPolicies, context canceled" client=459824:eu-central-1 module=aws-src table=aws_iot_thing_groups
2022-11-21T13:37:09Z ERR column resolver finished with error error="operation error IoT: ListTagsForResource, context canceled" client=459824:eu-central-1 module=aws-src table=aws_iot_thing_groups
2022-11-21T13:37:09Z ERR column resolver finished with error error="operation error IoT: ListThingPrincipals, request canceled, context canceled" client=459824:eu-central-1 module=aws-src table=aws_iot_things
2022-11-21T13:37:09Z INF Got interrupt. Source plugin server shutting down address=/tmp/cq-cptqnMDLFHDuEDpl.sock module=cli
2022-11-21T13:37:09Z ERR exiting with error error="failed to sync source aws: failed to fetch resources from stream: rpc error: code = ResourceExhausted desc = grpc: received message larger than max (9090290 vs. 4194304)" module=cli
```

**Comment by hermanschaaf** [comment]
Hi @altho1, thanks for the additional debug info. That helps a lot!

We've increased the max buffer size now, could you please check whether this resolves the issue and allows for a successful sync of your accounts?

You will need to be using the latest versions of the CLI as well as the source and destination plugins that were just released a couple of minutes ago:

- `cli v2.0.19`
- `aws v7.1.2`
- `postgresql v1.7.12`

**Comment by hermanschaaf** [comment]
Thanks for reporting back @altho1! I'll look into this some more, and hopefully have a solution soon.

For more context: this is a single row in a table (unclear which one) that's apparently taking up more than 50MiB. That's pretty big for a single row, and we likely wouldn't want rows as big as that in the database in any case. It could happen if a particular column with a custom resolver is making a lot of calls to the AWS API, but I think in this case we should reconsider how the resolver for that table is written, and possibly split it into separate tables.

To get to the bottom of this we'll need a bit more information from the logs. I'll make a change today that catches when we're trying to send a row bigger than the max, logs some details about it, and then drops it so the rest of the sync can at least succeed and we can investigate further.

**Comment by hermanschaaf** [comment]
Hi @altho1, thanks for that! 

We have a new set of releases again if you don't mind testing it out:

- cli [v2.0.20](https://github.com/cloudquery/cloudquery/compare/cli-v2.0.19...cli-v2.0.20)
- aws [v7.1.4](https://github.com/cloudquery/cloudquery/compare/plugins-source-aws-v7.1.3...plugins-source-aws-v7.1.4)
- postgresql [v1.7.13](https://github.com/cloudquery/cloudquery/compare/plugins-destination-postgresql-v1.7.12...plugins-destination-postgresql-v1.7.13)

This time I bumped the maximum to 100MiB, but we also now drop the message if it exceeds this limit. If you have telemetry turned on, we will get an anonymous report about which table hit this limit so we can investigate further (if you don't have telemetry turned on, a warning in the logs should also tell you which table it was--I'd appreciate if you could report back on that 😄 )

**Comment by altho1** [comment]
Hi @hermanschaaf,

Tested - not as expected though :)

Versions:
```
Downloading https://github.com/cloudquery/cloudquery/releases/download/plugins-source-aws-v7.1.4/aws_linux_amd64.zip
Downloading https://github.com/cloudquery/cloudquery/releases/download/plugins-destination-postgresql-v1.7.13/postgresql_linux_amd64.zip
```

Telemetry:
```
| Syncing resources... (894714/-, 1230 resources/s) [13m39s] 
                                                             

| Syncing resources... (894715/-, 1230 resources/s) [13m39s] 
                                                             
Error: failed to sync source aws: failed to fetch resources from stream: rpc error: code = ResourceExhausted desc = grpc: received message larger than max (56250590 vs. 52428800)
```

Here it starts (488840 lines down)
```
2022-11-22T12:08:23Z INF fetch table finished client=029532:us-east-1 errors=0 module=aws-src resources=0 table=aws_quicksight_analyses
2022-11-22T12:08:23Z INF table resolver started client=078428:us-east-2 module=aws-src table=aws_ram_resources
2022-11-22T12:08:23Z ERR table resolver finished with error error="operation error QuickSight: ListUsers, https response error StatusCode: 404, RequestID: f83a691b-0f75-4be7-aa5c-c351e50154a4, ResourceNotFoundException: Namespace default not found in account 093383. AWS account ID: 093383, Namespace: default." client=093383:us-east-1 module=aws-src table=aws_quicksight_users
2022-11-22T12:08:23Z INF fetch table finished client=093383:us-east-1 errors=1 module=aws-src resources=0 table=aws_quicksight_users
2022-11-22T12:08:23Z INF table resolver started client=078428:eu-north-1 module=aws-src table=aws_ram_resources
2022-11-22T12:08:24Z INF fetch table finished client=637734:ap-south-1 errors=0 module=aws-src resources=0 table=aws_quicksight_analyses
2022-11-22T12:08:24Z INF table resolver started client=078428:ap-northeast-3 module=aws-src table=aws_ram_resources
2022-11-22T12:08:24Z INF fetch table finished client=093383:eu-west-1 errors=0 module=aws-src resources=0 table=aws_neptune_clusters
2022-11-22T12:08:24Z INF table resolver started client=078428:ap-southeast-2 module=aws-src table=aws_ram_resources
2022-11-22T12:08:24Z ERR table resolver finished with error error="operation error QuickSight: ListUsers, https response error StatusCode: 404, RequestID: 843340ad-afbf-4a22-a363-b1d9899d458d, ResourceNotFoundException: Namespace default not found in account 831017. AWS account ID: 831017, Namespace: default." client=831017:ca-central-1 module=aws-src table=aws_quicksight_users
2022-11-22T12:08:24Z INF fetch table finished client=831017:ca-central-1 errors=1 module=aws-src resources=0 table=aws_quicksight_users
2022-11-22T12:08:24Z INF table resolver started client=078428:us-west-2 module=aws-src table=aws_ram_resources
2022-11-22T12:08:24Z INF fetch table finished client=165265:ap-northeast-2 errors=0 module=aws-src resources=187 table=aws_config_config_rule_compliances
2022-11-22T12:08:24Z ERR got error. failed to wait for plugin error="failed to execute batch: context canceled" module=cli
2022-11-22T12:08:24Z ERR finished call grpc.code=Unknown grpc.component=server grpc.error="failed to receive msg: rpc error: code = Canceled desc = context canceled" grpc.method=Write2 grpc.method_type=client_stream grpc.service=proto.Destination grpc.start_time=2022-11-22T11:54:43Z grpc.time_ms=820100.1 module=cli peer.address=@ protocol=grpc
2022-11-22T12:08:24Z INF Got interrupt. Destination plugin server shutting down address=/tmp/cq-WUVbsQkeefkXlChM.sock module=cli
2022-11-22T12:08:24Z ERR table resolver finished with error error="operation error QuickSight: ListDataSets, request canceled, context canceled" client=201409:ap-northeast-2 module=aws-src table=aws_quicksight_data_sets
2022-11-22T12:08:24Z INF fetch table finished client=316117:us-west-2 errors=0 module=aws-src resources=0 table=aws_neptune_cluster_snapshots
2022-11-22T12:08:24Z INF fetch table finished client=201409:ap-northeast-2 errors=1 module=aws-src resources=0 table=aws_quicksight_data_sets
2022-11-22T12:08:24Z ERR table resolver finished with error error="operation error QuickSight: ListUsers, request canceled, context canceled" client=029532:sa-east-1 module=aws-src table=aws_quicksight_users
2022-11-22T12:08:24Z ERR table resolver finished with error error="operation error QuickSight: ListGroups, request canceled, context canceled" client=316117:us-east-2 module=aws-src table=aws_quicksight_groups
2022-11-22T12:08:24Z INF fetch table finished client=316117:us-east-2 errors=1 module=aws-src resources=0 table=aws_quicksight_groups
2022-11-22T12:08:24Z ERR table resolver finished with error error="operation error Neptune: DescribeGlobalClusters, request canceled, context canceled" client=091888:eu-central-1 module=aws-src table=aws_neptune_global_clusters
2022-11-22T12:08:24Z INF fetch table finished client=091888:eu-central-1 errors=1 module=aws-src resources=0 table=aws_neptune_global_clusters
2022-11-22T12:08:24Z ERR table resolver finished with error error="operation error Neptune: DescribeDBClusters, request canceled, context canceled" client=182333:eu-west-2 module=aws-src table=aws_neptune_clusters
2022-11-22T12:08:24Z ERR table resolver finished with error error="operation error QuickSight: ListGroups, request canceled, context canceled" client=746744:ap-southeast-2 module=aws-src table=aws_quicksight_groups
2022-11-22T12:08:24Z INF fetch table finished client=182333:eu-west-2 errors=1 module=aws-src resources=0 table=aws_neptune_clusters
2022-11-22T12:08:24Z INF fetch table finished client=746744:ap-southeast-2 errors=1 module=aws-src resources=0 table=aws_quicksight_groups
2022-11-22T12:08:24Z ERR table resolver finished with error error="operation error Neptune: DescribeDBClusters, request canceled, context canceled" client=795105:eu-west-2 module=aws-src table=aws_neptune_clusters
2022-11-22T12:08:24Z INF fetch table finished client=795105:eu-west-2 errors=1 module=aws-src resources=0 table=aws_neptune_clusters
2022-11-22T12:08:24Z ERR table resolver finished with error error="operation error QuickSight: ListUsers, request canceled, context canceled" client=831017:sa-east-1 module=aws-src table=aws_quicksight_users
2022-11-22T12:08:24Z INF fetch table finished client=831017:sa-east-1 errors=1 module=aws-src resources=0 table=aws_quicksight_users
2022-11-22T12:08:24Z ERR table resolver finished with error error="operation error Neptune: DescribeDBSubnetGroups, request canceled, context canceled" client=310563:ap-northeast-1 module=aws-src table=aws_neptune_subnet_groups
2022-11-22T12:08:24Z INF fetch table finished client=310563:ap-northeast-1 errors=1 module=aws-src resources=0 table=aws_neptune_subnet_groups
2022-11-22T12:08:24Z ERR table resolver finished with error error="operation error RAM: ListPrincipals, request canceled, context canceled" client=795105:ap-southeast-2 module=aws-src table=aws_ram_principals
2022-11-22T12:08:24Z INF fetch table finished client=795105:ap-southeast-2 errors=1 module=aws-src resources=0 table=aws_ram_principals
2022-11-22T12:08:24Z ERR table resolver finished with error error="operation error Neptune: DescribeDBClusters, request canceled, context canceled" client=664354:eu-central-1 module=aws-src table=aws_neptune_clusters
2022-11-22T12:08:24Z INF fetch table finished client=664354:eu-central-1 errors=1 module=aws-src resources=0 table=aws_neptune_clusters
2022-11-22T12:08:24Z ERR table resolver finished with error error="operation error Neptune: DescribeDBParameterGroups, request canceled, context canceled" client=521949:us-east-1 module=aws-src table=aws_neptune_db_parameter_groups
2022-11-22T12:08:24Z ERR table resolver finished with error error="operation error QuickSight: ListTemplates, request canceled, context canceled" client=459824:ap-northeast-2 module=aws-src table=aws_quicksight_templates
2022-11-22T12:08:24Z INF fetch table finished client=521949:us-east-1 errors=1 module=aws-src resources=0 table=aws_neptune_db_parameter_groups
```


how it ends (501446 lines down)
```

2022-11-22T12:08:24Z ERR table resolver finished with error error="operation error Neptune: DescribeDBClusterParameterGroups, request canceled, context canceled" client=664354:eu-central-1 module=aws-src table=aws_neptune_cluster_parameter_groups
2022-11-22T12:08:24Z INF fetch table finished client=664354:eu-central-1 errors=1 module=aws-src resources=0 table=aws_neptune_cluster_parameter_groups
2022-11-22T12:08:24Z ERR table resolver finished with error error="operation error Neptune: DescribeDBSubnetGroups, request canceled, context canceled" client=664354:ap-northeast-1 module=aws-src table=aws_neptune_subnet_groups
2022-11-22T12:08:24Z INF fetch table finished client=664354:ap-northeast-1 errors=1 module=aws-src resources=0 table=aws_neptune_subnet_groups
2022-11-22T12:08:24Z INF fetch table finished client=919674:ap-northeast-2 errors=0 module=aws-src resources=0 table=aws_neptune_cluster_snapshots
2022-11-22T12:08:24Z ERR table resolver finished with error error="operation error Neptune: DescribeDBSubnetGroups, request canceled, context canceled" client=745427:ap-northeast-1 module=aws-src table=aws_neptune_subnet_groups
2022-11-22T12:08:24Z INF fetch table finished client=745427:ap-northeast-1 errors=1 module=aws-src resources=0 table=aws_neptune_subnet_groups
2022-11-22T12:08:24Z ERR table resolver finished with error error="operation error Neptune: DescribeEventSubscriptions, request canceled, context canceled" client=569746:us-east-2 module=aws-src table=aws_neptune_event_subscriptions
2022-11-22T12:08:24Z INF fetch table finished client=569746:us-east-2 errors=1 module=aws-src resources=0 table=aws_neptune_event_subscriptions
2022-11-22T12:08:24Z ERR table resolver finished with error error="operation error Neptune: DescribeDBClusters, request canceled, context canceled" client=470456:eu-central-1 module=aws-src table=aws_neptune_clusters
2022-11-22T12:08:24Z INF fetch table finished client=470456:eu-central-1 errors=1 module=aws-src resources=0 table=aws_neptune_clusters
2022-11-22T12:08:24Z ERR table resolver finished with error error="operation error Neptune: DescribeGlobalClusters, request canceled, context canceled" client=316117:us-west-2 module=aws-src table=aws_neptune_global_clusters
2022-11-22T12:08:24Z INF fetch table finished client=316117:us-west-2 errors=1 module=aws-src resources=0 table=aws_neptune_global_clusters
2022-11-22T12:08:24Z ERR table resolver finished with error error="operation error Neptune: DescribeDBClusterParameterGroups, request canceled, context canceled" client=470456:eu-central-1 module=aws-src table=aws_neptune_cluster_parameter_groups
2022-11-22T12:08:24Z INF fetch table finished client=470456:eu-central-1 errors=1 module=aws-src resources=0 table=aws_neptune_cluster_parameter_groups
2022-11-22T12:08:24Z ERR table resolver finished with error error="operation error Neptune: DescribeDBClusters, request canceled, context canceled" client=119311:eu-north-1 module=aws-src table=aws_neptune_clusters
2022-11-22T12:08:24Z INF fetch table finished client=119311:eu-north-1 errors=1 module=aws-src resources=0 table=aws_neptune_clusters
2022-11-22T12:08:24Z ERR table resolver finished with error error="operation error Neptune: DescribeDBClusterParameterGroups, request canceled, context canceled" client=059487:us-east-2 module=aws-src table=aws_neptune_cluster_parameter_groups
2022-11-22T12:08:24Z INF fetch table finished client=059487:us-east-2 errors=1 module=aws-src resources=0 table=aws_neptune_cluster_parameter_groups
2022-11-22T12:08:24Z ERR table resolver finished with error error="operation error Neptune: DescribeDBClusterParameterGroups, request canceled, context canceled" client=521949:us-east-2 module=aws-src table=aws_neptune_cluster_parameter_groups
2022-11-22T12:08:24Z INF fetch table finished client=521949:us-east-2 errors=1 module=aws-src resources=0 table=aws_neptune_cluster_parameter_groups
2022-11-22T12:08:24Z INF fetch table finished client=159196:ap-northeast-1 errors=1 module=aws-src resources=0 table=aws_quicksight_data_sources
2022-11-22T12:08:24Z INF Got interrupt. Source plugin server shutting down address=/tmp/cq-clfHIHdZTBkcABdi.sock module=cli
2022-11-22T12:08:24Z INF fetch table finished client=795105:ap-northeast-1 errors=0 module=aws-src resources=0 table=aws_neptune_cluster_snapshots
2022-11-22T12:08:24Z INF fetch table finished client=182333:ap-northeast-1 errors=1 module=aws-src resources=0 table=aws_ram_resources
2022-11-22T12:08:24Z ERR table resolver finished with error error="operation error RAM: ListPrincipals, request canceled, context canceled" client=116928:eu-north-1 module=aws-src table=aws_ram_principals
2022-11-22T12:08:24Z INF fetch table finished client=116928:eu-north-1 errors=1 module=aws-src resources=0 table=aws_ram_principals
2022-11-22T12:08:24Z INF fetch table finished client=795105:ap-northeast-2 errors=0 module=aws-src resources=0 table=aws_neptune_cluster_snapshots
2022-11-22T12:08:24Z ERR table resolver finished with error error="operation error Neptune: DescribeGlobalClusters, request canceled, context canceled" client=121995:ap-northeast-2 module=aws-src table=aws_neptune_global_clusters
2022-11-22T12:08:24Z INF fetch table finished client=121995:ap-northeast-2 errors=1 module=aws-src resources=0 table=aws_neptune_global_clusters
2022-11-22T12:08:24Z INF table resolver started client=637734:ap-southeast-2 module=aws-src table=aws_ram_resources
2022-11-22T12:08:24Z ERR exiting with error error="failed to sync source aws: failed to fetch resources from stream: rpc error: code = ResourceExhausted desc = grpc: received message larger than max (56250590 vs. 52428800)" module=cli
```

**Comment by hermanschaaf** [comment]
@altho1 Oh no 😮‍💨 Just checking, did you also use the latest version of the CLI (v2.0.20) here?

**Comment by altho1** [comment]
@hermanschaaf I did, and wont share the result - just the logs from the next run :-D

**Comment by altho1** [comment]
@hermanschaaf It is now running on expected versions.

I have the telemetry data in CW and can search, but cannot find "Large", should I search for something else?
When the batch process has run (normally takes ~8 hours), then I will have the debug log to share.

For now it looks much better:
```
- Syncing resources... (11742528/-, 653 resources/s) [1h44m2s] 
                                                                 

\ Syncing resources... (11742529/-, 653 resources/s) [1h44m2s]
```

**Comment by hermanschaaf** [comment]
@altho1 Great! Maybe the 100MiB limit was enough to get it over the line, in which case you won't see any log lines about it. But can search for a warning stating `Row exceeding max bytes ignored`?

**Comment by altho1** [comment]
@hermanschaaf Sure, but that message is for when it passes 100mb and we have only seen it pass 50mb.
Last run it failed after ~13m and < 1mio resources. It has now passed 1,5hrs and 11mio resources, but has not reached the 100mb/2 mark that you are also reporting. Shouldn't I have seen that by now or will it only show up in the log and not the telemetry?

**Comment by hermanschaaf** [comment]
@altho1 Right, it's possible there won't be any log lines now. If you have telemetry enabled (i.e. `CQ_TELEMETRY_LEVEL` or `--telemetry-level` is not `none`) then our Sentry server will get a report if a message exceeds 50MiB. Warnings are only logged for messages over 100MiB, because at that point we drop it now. But we haven't received a Sentry event yet, so I assume telemetry is disabled. Perhaps I should have kept a log line for everything over 50MiB, but at least it sounds like it's getting along better now, so that's good :)

**Comment by altho1** [comment]
@hermanschaaf Hmm, isn't the default all? I do not have `--telemetry-level` set to `none`, and as you can see it has internet access (downloading plugins).
I will run the nightly batch with it explicitly set to `all`, just to be sure. Let's see what that brings us.

**Comment by altho1** [comment]
HI @hermanschaaf, 

We are closing in :)

Versions
```
@bc49f0e76ee0:~$ ./cloudquery --version
cloudquery version 2.0.20
Downloading https://github.com/cloudquery/cloudquery/releases/download/plugins-source-aws-v7.1.4/aws_linux_amd64.zip
Downloading https://github.com/cloudquery/cloudquery/releases/download/plugins-destination-postgresql-v1.7.13/postgresql_linux_amd64.zip
```

Console stats:
```
\ Syncing resources... (15892551/-, 27 resources/s) [6h0m43s] 
                                                                 
Sync completed successfully. Resources: 15892551, Errors: 12924, Panics: 0, Time: 6h1m58s
```

Random errors:
```
2022-11-23T00:13:06Z ERR table resolver finished with error error="operation error ECR PUBLIC: DescribeRepositories, https response error StatusCode: 400, RequestID: e61dc0d6-02dc-41e1-8599-56d741caa557, api error UnsupportedCommandException: DescribeRepositories command is only supported in us-east-1." client=093383:us-west-2 module=aws-src table=aws_ecrpublic_repositories
2022-11-23T00:13:06Z ERR table resolver finished with error error="operation error CloudTrail: DescribeTrails, exceeded maximum number of attempts, 10, https response error StatusCode: 0, RequestID: , request send failed, Post \"https://cloudtrail.ap-southeast-1.amazonaws.com/\": dial tcp: lookup cloudtrail.ap-southeast-1.amazonaws.com on xxx.xxx.0.2:53: dial udp xxx.xxx.0.2:53: socket: too many open files" client=459824:ap-southeast-1 module=aws-src table=aws_cloudtrail_trails
2022-11-23T00:13:06Z ERR table resolver finished with error error="operation error CloudHSM V2: DescribeClusters, exceeded maximum number of attempts, 10, https response error StatusCode: 0, RequestID: , request send failed, Post \"https://cloudhsmv2.eu-west-1.amazonaws.com/\": dial tcp: lookup cloudhsmv2.eu-west-1.amazonaws.com on xxx.xxx.0.2:53: dial udp xxx.xxx.0.2:53: socket: too many open files" client=323930:eu-west-1 module=aws-src table=aws_cloudhsmv2_clusters
2022-11-23T00:13:07Z ERR table resolver finished with error error="operation error Cognito Identity Provider: ListUserPools, exceeded maximum number of attempts, 10, https response error StatusCode: 0, RequestID: , request send failed, Post \"https://cognito-idp.ap-south-1.amazonaws.com/\": dial tcp [2406:da1a:93d:8501:xxxx:xxxx:xxxx:xxxx]:443: connect: network is unreachable" client=784246:ap-south-1 module=aws-src table=aws_cognito_user_pools
```

Here it takes a long time to complete, and then errors start:
```
2022-11-23T01:57:04Z INF table resolver started client=438071:eu-west-1 module=aws-src table=aws_glue_database_tables
2022-11-23T01:57:04Z INF fetch table finished client=438071:eu-west-1 errors=0 module=aws-src resources=15373 table=aws_glue_database_tables
2022-11-23T01:57:04Z INF table resolver started client=438071:eu-west-1 module=aws-src table=aws_glue_database_tables
2022-11-23T01:57:05Z INF table resolver started client=438071:eu-west-1 module=aws-src table=aws_glue_database_table_indexes
2022-11-23T01:57:05Z INF fetch table finished client=438071:eu-west-1 errors=0 module=aws-src resources=0 table=aws_glue_database_table_indexes
2022-11-23T01:57:05Z INF fetch table finished client=438071:eu-west-1 errors=0 module=aws-src resources=15374 table=aws_glue_database_tables
2022-11-23T01:57:05Z INF fetch table finished client=438071:eu-west-1 errors=0 module=aws-src resources=3 table=aws_glue_databases
2022-11-23T02:01:41Z INF fetch table finished client=119311:eu-central-1 errors=0 module=aws-src resources=318112 table=aws_kms_key_grants
2022-11-23T02:01:41Z INF fetch table finished client=119311:eu-central-1 errors=0 module=aws-src resources=13 table=aws_kms_keys
2022-11-23T02:15:37Z INF fetch table finished client=459824:ap-northeast-2 errors=0 module=aws-src resources=25740 table=aws_iot_things
2022-11-23T02:37:31Z INF fetch table finished client=182333:us-east-1 errors=0 module=aws-src resources=35863 table=aws_iot_things
2022-11-23T02:52:39Z INF fetch table finished client=059487:eu-central-1 errors=0 module=aws-src resources=62546 table=aws_iot_certificates
2022-11-23T02:57:19Z INF fetch table finished client=459824:ap-southeast-2 errors=0 module=aws-src resources=45405 table=aws_iot_things
2022-11-23T03:01:54Z INF fetch table finished client=059487:eu-west-1 errors=0 module=aws-src resources=61727 table=aws_iot_certificates
2022-11-23T03:07:44Z INF fetch table finished client=459824:ap-northeast-2 errors=0 module=aws-src resources=26065 table=aws_iot_certificates
2022-11-23T03:07:48Z INF fetch table finished client=059487:eu-west-1 errors=0 module=aws-src resources=61769 table=aws_iot_things
2022-11-23T03:33:50Z INF fetch table finished client=059487:ap-southeast-2 errors=0 module=aws-src resources=61697 table=aws_iot_things
2022-11-23T03:34:55Z INF fetch table finished client=459824:eu-central-1 errors=0 module=aws-src resources=95518 table=aws_iot_certificates
2022-11-23T03:45:54Z INF fetch table finished client=182333:us-east-1 errors=0 module=aws-src resources=74779 table=aws_iot_policies
2022-11-23T03:46:10Z INF fetch table finished client=059487:eu-central-1 errors=0 module=aws-src resources=62608 table=aws_iot_things
2022-11-23T03:58:34Z INF fetch table finished client=459824:eu-west-1 errors=0 module=aws-src resources=99140 table=aws_iot_certificates
2022-11-23T04:11:28Z INF fetch table finished client=459824:eu-west-1 errors=0 module=aws-src resources=99258 table=aws_iot_things
2022-11-23T04:28:05Z INF fetch table finished client=459824:us-east-1 errors=0 module=aws-src resources=126015 table=aws_iot_certificates
2022-11-23T04:47:19Z INF fetch table finished client=059487:ap-northeast-2 errors=0 module=aws-src resources=92982 table=aws_iot_things
2022-11-23T04:49:21Z INF fetch table finished client=459824:ap-northeast-2 errors=0 module=aws-src resources=50543 table=aws_iot_policies
2022-11-23T05:09:50Z INF fetch table finished client=459824:ap-southeast-2 errors=0 module=aws-src resources=50980 table=aws_iot_certificates
2022-11-23T05:41:22Z INF fetch table finished client=459824:eu-central-1 errors=0 module=aws-src resources=110993 table=aws_iot_things
2022-11-23T05:47:39Z INF fetch table finished client=059487:eu-central-1 errors=0 module=aws-src resources=127012 table=aws_iot_policies
2022-11-23T05:48:03Z INF fetch table finished client=459824:us-east-1 errors=0 module=aws-src resources=122310 table=aws_iot_things
2022-11-23T05:58:22Z INF fetch table finished client=059487:ap-southeast-2 errors=0 module=aws-src resources=61693 table=aws_iot_certificates
2022-11-23T06:07:14Z ERR column resolver finished with error error="operation error IoT: ListTagsForResource, failed to sign request: failed to retrieve credentials: failed to refresh cached credentials, operation error STS: AssumeRole, https response error StatusCode: 403, RequestID: 4415c3fc-eb88-4b2d-b955-1dbe40693f78, api error ExpiredToken: The security token included in the request is expired" client=059487:eu-west-1 module=aws-src table=aws_iot_policies
2022-11-23T06:07:14Z ERR table resolver finished with error error="operation error IoT: GetPolicy, failed to sign request: failed to retrieve credentials: failed to refresh cached credentials, operation error STS: AssumeRole, https response error StatusCode: 403, RequestID: 4415c3fc-eb88-4b2d-b955-1dbe40693f78, api error ExpiredToken: The security token included in the request is expired" client=059487:eu-west-1 module=aws-src table=aws_iot_policies
2022-11-23T06:07:14Z INF fetch table finished client=059487:eu-west-1 errors=2 module=aws-src resources=121169 table=aws_iot_policies
2022-11-23T06:07:14Z ERR column resolver finished with error error="operation error IoT: ListAttachedPolicies, failed to sign request: failed to retrieve credentials: failed to refresh cached credentials, operation error STS: AssumeRole, https response error StatusCode: 403, RequestID: 35e60035-f79c-4449-9ee7-cf4a2931eafd, api error ExpiredToken: The security token included in the request is expired" client=059487:us-east-1 module=aws-src table=aws_iot_certificates
2022-11-23T06:07:14Z ERR table resolver finished with error error="operation error IoT: DescribeCertificate, failed to sign request: failed to retrieve credentials: failed to refresh cached credentials, operation error STS: AssumeRole, https response error StatusCode: 403, RequestID: 35e60035-f79c-4449-9ee7-cf4a2931eafd, api error ExpiredToken: The security token included in the request is expired" client=059487:us-east-1 module=aws-src table=aws_iot_certificates
2022-11-23T06:07:14Z INF fetch table finished client=059487:us-east-1 errors=2 module=aws-src resources=198702 table=aws_iot_certificates
2022-11-23T06:07:14Z ERR column resolver finished with error error="operation error IoT: ListThingPrincipals, failed to sign request: failed to retrieve credentials: failed to refresh cached credentials, operation error STS: AssumeRole, https response error StatusCode: 403, RequestID: 7a93b66a-a226-4150-8a73-06e24c5b4a14, api error ExpiredToken: The security token included in the request is expired" client=059487:us-east-1 module=aws-src table=aws_iot_things
2022-11-23T06:07:14Z ERR column resolver finished with error error="operation error IoT: ListTagsForResource, failed to sign request: failed to retrieve credentials: failed to refresh cached credentials, operation error STS: AssumeRole, https response error StatusCode: 403, RequestID: 09ba667f-8023-4789-bb26-7aae226dcd5f, api error ExpiredToken: The security token included in the request is expired" client=059487:ap-southeast-2 module=aws-src table=aws_iot_policies
2022-11-23T06:07:14Z ERR table resolver finished with error error="operation error IoT: GetPolicy, failed to sign request: failed to retrieve credentials: failed to refresh cached credentials, operation error STS: AssumeRole, https response error StatusCode: 403, RequestID: 09ba667f-8023-4789-bb26-7aae226dcd5f, api error ExpiredToken: The security token included in the request is expired" client=059487:ap-southeast-2 module=aws-src table=aws_iot_policies
2022-11-23T06:07:14Z INF fetch table finished client=059487:ap-southeast-2 errors=2 module=aws-src resources=63699 table=aws_iot_policies
2022-11-23T06:07:14Z ERR table resolver finished with error error="operation error IoT: DescribeCertificate, failed to sign request: failed to retrieve credentials: failed to refresh cached credentials, operation error STS: AssumeRole, https response error StatusCode: 403, RequestID: 38322e4f-1d73-47cb-8f5b-6934a10ca5f3, api error ExpiredToken: The security token included in the request is expired" client=059487:ap-northeast-2 module=aws-src table=aws_iot_certificates
2022-11-23T06:07:14Z ERR column resolver finished with error error="operation error IoT: ListAttachedPolicies, failed to sign request: failed to retrieve credentials: failed to refresh cached credentials, operation error STS: AssumeRole, https response error StatusCode: 403, RequestID: 38322e4f-1d73-47cb-8f5b-6934a10ca5f3, api error ExpiredToken: The security token included in the request is expired" client=059487:ap-northeast-2 module=aws-src table=aws_iot_certificates
2022-11-23T06:07:14Z INF fetch table finished client=059487:ap-northeast-2 errors=2 module=aws-src resources=70026 table=aws_iot_certificates
2022-11-23T06:07:14Z ERR table resolver finished with error error="operation error IoT: GetPolicy, failed to sign request: failed to retrieve credentials: failed to refresh cached credentials, operation error STS: AssumeRole, https response error StatusCode: 403, RequestID: e49ae815-d072-4542-a451-cd8cae8f9510, api error ExpiredToken: The security token included in the request is expired" client=059487:ap-northeast-2 module=aws-src table=aws_iot_policies
2022-11-23T06:07:14Z ERR column resolver finished with error error="operation error IoT: ListTagsForResource, failed to sign request: failed to retrieve credentials: failed to refresh cached credentials, operation error STS: AssumeRole, https response error StatusCode: 403, RequestID: e49ae815-d072-4542-a451-cd8cae8f9510, api error ExpiredToken: The security token included in the request is expired" client=059487:ap-northeast-2 module=aws-src table=aws_iot_policies
2022-11-23T06:07:14Z INF fetch table finished client=059487:ap-northeast-2 errors=2 module=aws-src resources=69782 table=aws_iot_policies
2022-11-23T06:07:14Z ERR column resolver finished with error error="operation error IoT: ListThingPrincipals, failed to sign request: failed to retrieve credentials: failed to refresh cached credentials, operation error STS: AssumeRole, https response error StatusCode: 403, RequestID: 6a2571d4-ea4f-431d-9d7a-21df88afbf9b, api error ExpiredToken: The security token included in the request is expired" client=059487:us-east-1 module=aws-src table=aws_iot_things
2022-11-23T06:07:14Z ERR column resolver finished with error error="operation error IoT: ListThingPrincipals, failed to sign request: failed to retrieve credentials: failed to refresh cached credentials, operation error STS: AssumeRole, https response error StatusCode: 403, RequestID: 460e51a3-524c-485e-8aad-b3150bf5a4bd, api error ExpiredToken: The security token included in the request is expired" client=059487:us-east-1 module=aws-src table=aws_iot_things
2022-11-23T06:07:14Z ERR column resolver finished with error error="operation error IoT: ListThingPrincipals, failed to sign request: failed to retrieve credentials: failed to refresh cached credentials, operation error STS: AssumeRole, https response error StatusCode: 403, RequestID: 9ab78478-ffd4-4d54-b31d-95462b66b851, api error ExpiredToken: The security token included in the request is expired" client=059487:us-east-1 module=aws-src table=aws_iot_things
2022-11-23T06:07:14Z ERR column resolver finished with error error="operation error IoT: ListThingPrincipals, failed to sign request: failed to retrieve credentials: failed to refresh cached credentials, operation error STS: AssumeRole, https response error StatusCode: 403, RequestID: 00e50b7c-c372-43a0-a97c-f7d13ec6fe62, api error ExpiredToken: The security token included in the request is expired" client=059487:us-east-1 module=aws-src table=aws_iot_things
2022-11-23T06:07:14Z ERR column resolver finished with error error="operation error IoT: ListThingPrincipals, failed to sign request: failed to retrieve credentials: failed to refresh cached credentials, operation error STS: AssumeRole, https response error StatusCode: 403, RequestID: 00e50b7c-c372-43a0-a97c-f7d13ec6fe62, api error ExpiredToken: The security token included in the request is expired" client=059487:us-east-1 module=aws-src table=aws_iot_things
```

and here it ends:
```
2022-11-23T06:07:49Z ERR column resolver finished with error error="operation error IoT: ListThingPrincipals, failed to sign request: failed to retrieve credentials: failed to refresh cached credentials, operation error STS: AssumeRole, https response error StatusCode: 403, RequestID: 652403a0-377b-4004-b640-e49082e147b9, api error ExpiredToken: The security token included in the request is expired" client=784246:eu-west-1 module=aws-src table=aws_iot_things
2022-11-23T06:07:49Z ERR column resolver finished with error error="operation error IoT: ListThingPrincipals, failed to sign request: failed to retrieve credentials: failed to refresh cached credentials, operation error STS: AssumeRole, https response error StatusCode: 403, RequestID: 652403a0-377b-4004-b640-e49082e147b9, api error ExpiredToken: The security token included in the request is expired" client=784246:eu-west-1 module=aws-src table=aws_iot_things
2022-11-23T06:07:49Z INF fetch table finished client=784246:eu-west-1 errors=346 module=aws-src resources=172250 table=aws_iot_things
2022-11-23T06:07:49Z INF sync finished duration=21608705.639348 errors=12924 module=aws-src panics=0 resources=15892551
2022-11-23T06:07:49Z INF finished call grpc.code=OK grpc.component=server grpc.method=Sync2 grpc.method_type=server_stream grpc.service=proto.Source grpc.start_time=2022-11-23T00:07:06Z grpc.time_ms=2.164313e+07 module=cli peer.address=@ protocol=grpc
2022-11-23T06:08:39Z INF finished call grpc.code=OK grpc.component=server grpc.method=Write2 grpc.method_type=client_stream grpc.service=proto.Destination grpc.start_time=2022-11-23T00:07:06Z grpc.time_ms=2.1693308e+07 module=cli peer.address=@ protocol=grpc
2022-11-23T06:08:39Z INF started call grpc.code=OK grpc.component=server grpc.method=Close grpc.method_type=unary grpc.service=proto.Destination grpc.start_time=2022-11-23T06:08:39Z grpc.time_ms=0.02 module=cli peer.address=@ protocol=grpc
2022-11-23T06:08:39Z INF finished call grpc.code=OK grpc.component=server grpc.method=Close grpc.method_type=unary grpc.service=proto.Destination grpc.start_time=2022-11-23T06:08:39Z grpc.time_ms=0.129 module=cli peer.address=@ protocol=grpc
2022-11-23T06:08:39Z INF started call grpc.code=OK grpc.component=server grpc.method=GetMetrics grpc.method_type=unary grpc.service=proto.Source grpc.start_time=2022-11-23T06:08:39Z grpc.time_ms=0.021 module=cli peer.address=@ protocol=grpc
2022-11-23T06:08:40Z INF finished call grpc.code=OK grpc.component=server grpc.method=GetMetrics grpc.method_type=unary grpc.service=proto.Source grpc.start_time=2022-11-23T06:08:39Z grpc.time_ms=12.048 module=cli peer.address=@ protocol=grpc
2022-11-23T06:08:40Z INF Sending sync summary to analyticsv1.cloudquery.io:443 module=cli
2022-11-23T06:08:40Z INF Got interrupt. Destination plugin server shutting down address=/tmp/cq-qXwMEuuFYmTxXqRz.sock module=cli
2022-11-23T06:08:40Z INF End sync destinations=["postgresql"] module=cli source=aws sync_time=2022-11-23T00:06:41Z
2022-11-23T06:08:40Z INF Got interrupt. Source plugin server shutting down address=/tmp/cq-NMfYcFwUDqrMaqOZ.sock module=cli
```

I would expect to see a warning for passing the 50mb limit, as my data hasn't changed much?
Did you get sentry entries?

**Comment by hermanschaaf** [comment]
@altho1 Ah, great, glad it completed this time! The remaining errors look unrelated to the original gRPC issue, but we can take a look. Some number of errors is expected on a large cloud estate like this one, but the `failed to refresh cached credentials` for IOT doesn't look right. I'll raise a new issue for that one.

For the `socket: too many open files` error, you may want to try two things:
 - checking that the open file limit on the system is as high as it can be (`ulimit -a` to check the limits, `ulimit -n unlimited` to set it, but this is somewhat system dependant)
 - tuning the `concurrency` option down a little bit. It defaults to 1,000,000 https://www.cloudquery.io/docs/reference/source-spec#concurrency (also https://www.cloudquery.io/docs/advanced-topics/rate-limiting). 

The logs are only written over 100MiB, so it's possible that we wouldn't have seen one. I did expect to get a Sentry entry though, and didn't. I think it's likely that an inbound filter is accidentally filtering it out on the Sentry side, so we'll take a look at that.

By the way, feel free to hop onto our [Discord server](https://www.cloudquery.io/discord) if you'd like to discuss any of these issues in more detail / semi-realtime 😄

**Comment by altho1** [comment]
Thanx @hermanschaaf,

You can close this one.

My ulimit is at max (https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_Ulimit.html)
I will turn down the concurrency - nofiles max is 4096 - should the concurrency be half or less ?

I am on Discord - thanx

**Comment by hermanschaaf** [comment]
@altho1 Concurrency doesn't exactly match number of open files, it's closer to number of concurrent goroutines. I think if you were to tune it down to ~4000 the sync would slow to a crawl. I'd suggest: 
 - making sure in this case that the `nofile` soft limit is the same as the hard limit (4096)
 - tuning concurrency down a little bit, maybe to 500,000 at first to see if that helps. I wouldn't suggest going lower than 100,000, as then it'd probably get too slow. If the file limits are still a blocker at that point it might be better to run multiple containers in parallel (with each container syncing a subset of tables or accounts, perhaps). This works fine as long as each source config `name` entry is unique.

**Comment by hermanschaaf** [comment]
In any case, it seems like the original issue is solved, so I'll close this issue. Thanks for reporting, sharing the logs and trying the new versions! Feel free to reach out on Discord or in a new issue if you'd like to discuss something further 😄
